src/modules/cmdpal/Microsoft.CmdPal.UI/ShellPage.xaml

```diff
@@ -7,10 +7,49 @@
     xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
     xmlns:local="using:Microsoft.CmdPal.UI"
     xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
+    xmlns:toolkit="using:CommunityToolkit.WinUI.UI.Controls"
+    xmlns:converters="using:CommunityToolkit.WinUI.Converters"
     Background="Transparent"
     mc:Ignorable="d">
 
+    <Page.Resources>
+        <ResourceDictionary>
+            <converters:StringVisibilityConverter
+                x:Key="StringNotEmptyToVisibilityConverter"
+                EmptyValue="Collapsed"
+                NotEmptyValue="Visible" />
+            <converters:BoolNegationConverter x:Key="BoolNegationConverter" />
+        </ResourceDictionary>
+    </Page.Resources>
+
     <Grid>
+
+        <VisualStateManager.VisualStateGroups>
+            <VisualStateGroup>
+
+                <VisualState x:Name="DetailsCollapsed">
+                    <VisualState.StateTriggers>
+                        <StateTrigger IsActive="{x:Bind ViewModel.IsDetailsVisible, Mode=OneWay, Converter={StaticResource BoolNegationConverter}}"/>
+                    </VisualState.StateTriggers>
+                    <VisualState.Setters>
+                        <Setter Target="DetailsContent.Visibility" Value="Collapsed" />
+                        <Setter Target="DetailsColumn.Width" Value="Auto" />
+                    </VisualState.Setters>
+                </VisualState>
```

Comment:
```suggestion
                <VisualState x:Name="DetailsCollapsed" />
```
These are all the default values in XAML, so you don't need this. the `DetailsVisible` state will trigger when the property IsActive and then override the values, when that's no longer the case they'll automatically go back to what they were. (This also means we don't need the converter anymore)

https://learn.microsoft.com/windows/windows-app-sdk/api/winrt/microsoft.ui.xaml.visualstate#default-state

Reply:
Oh that's neat